### PR TITLE
Hide dummy recipes from Elven NEI handler

### PR DESCRIPTION
**Pull request by miozune**
like iron -> iron

### CODE REVIEW

File: repositories.gradle [hunk 1]
```diff
@@ -7,6 +7,7 @@ repositories {
     maven {
         name 'ic2'
         url 'https://maven.ic2.player.to/'
+        url = "https://maven2.ic2.player.to/"
```
Comment (combusterf) [inline repositories.gradle:10]: This change doesn't look relevant for this MR (and its inconsistent code style). Can you revert this?

File: src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerElvenTrade.java [hunk 2]
```diff
@@ -152,4 +153,28 @@ public void loadUsageRecipes(ItemStack ingredient) {
...
+	// hide dummy recipes
+	private List<RecipeElvenTrade> filteredElvenTradeRecipes() {
+		List<RecipeElvenTrade> list = new ArrayList<>();
+		for (RecipeElvenTrade recipe : BotaniaAPI.elvenTradeRecipes) {
+			if (recipe == null)
+				continue;
+
+			if (recipe.getInputs().size() == 1 && stackSame(recipe.getOutput(), recipe.getInputs().get(0))) {
+				continue;
+			}
```
Comment (combusterf) [inline src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerElvenTrade.java:165]: Being consistent with the braces would be nice.

**Review by combusterf — COMMENTED** [review]
Just a few things that would be appreciated to make the code cleaner. Looks good in terms of functionality.

File: repositories.gradle (hunk 1 above)
Comment (kubasz) [reply repositories.gradle:10]: This is important, the main ic2 maven mirror is down and the second one should be added to all repos going forward to keep them possible to build

Comment (kubasz) [reply repositories.gradle:10]: But yeah, I don't think the `=` is required in Groovy syntax

File: src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerElvenTrade.java [hunk 3]
```diff
@@ -152,4 +153,28 @@ public void loadUsageRecipes(ItemStack ingredient) {
 		}
 	}
 
+	// hide dummy recipes
+	private List<RecipeElvenTrade> filteredElvenTradeRecipes() {
+		List<RecipeElvenTrade> list = new ArrayList<>();
```
Comment (Glease) [inline src/main/java/vazkii/botania/client/integration/nei/recipe/RecipeHandlerElvenTrade.java:158]: It'd be nice to not copy the list on every invocation. Maybe use java Stream API or use guava's Iterables.filter. Either is OK imo. The first is the standard going forward, while the second offers the friendly Iterable interface so you can use it in a for each structrue as usual.

File: repositories.gradle (hunk 1 above)
Comment (miozune) [reply repositories.gradle]: my IDEA goes mad, compilation works though

**Pull request merged**